### PR TITLE
Migrate top Gitlab agent

### PR DESCRIPTION
**Pull request by veidenberg**
### Description
Migrate deployment jobs to Gitlab agent.

### Related JIRA Issue(s)
[ENSWBSITES-2489](https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2489)

### Review App URL(s) 
http://gitlab-agent.review.ensembl.org

### CODE REVIEW

File: backend-server/.backend-server-ci.yml [hunk 1]
```diff
@@ -100,42 +103,59 @@ BE-Docker:
...
+    name: production
+  variables:
+    AGENT: ${PROD_AGENT}
+    NS: ${PROS_NS}
 
 Production:WP41:HX:
   extends: .deploy_backend
   environment:
-    name: wp41-hx-prod
+    name: production
```
Comment (jonk-ebi) [inline backend-server/.backend-server-ci.yml:114]: I believe this should use the fallback environment like this https://github.com/Ensembl/ensembl-web-metadata-api/blob/jk_gitlab_agent_migration/.gitlab-ci.yml#L127

File: backend-server/.backend-server-ci.yml [hunk 2]
```diff
@@ -100,42 +103,59 @@ BE-Docker:
 Production:WP40:HL:
```
Comment (jonk-ebi) [inline backend-server/.backend-server-ci.yml:103]: On the example PR the cluster name was scrubbed from the stage name

**Review by jonk-ebi — CHANGES_REQUESTED** [review]

File: backend-server/.backend-server-ci.yml [hunk 3]
```diff
@@ -100,42 +103,59 @@ BE-Docker:
...
+    NS: ${PROS_NS}
 
 Production:WP41:HX:
   extends: .deploy_backend
   environment:
-    name: wp41-hx-prod
+    name: production
+  variables:
+    AGENT: ${PROD_AGENT}
+    NS: ${PROS_NS}
```
Comment (kamaldodiya) [inline backend-server/.backend-server-ci.yml:117]: Fallback agent and ns see the section here  Jobname + variable name
https://gitlab.ebi.ac.uk/ensembl-web/ensembl_search_hub/-/blob/master/.gitlab-ci.yml#L36-L43

**Review by kamaldodiya — CHANGES_REQUESTED** [review]
Change Job name + variables

**Review by kamaldodiya — APPROVED** [review]

**Review by jonk-ebi — APPROVED** [review]

**Review by kamaldodiya — APPROVED** [review]

**Pull request merged**